src/network/taraxa_capability.cpp

```diff
@@ -1074,9 +1074,10 @@ void TaraxaCapability::sendSyncedMessage() {
   }
 }
 
-void TaraxaCapability::onNewBlockVerified(DagBlock const &block) {
-  // If node is syncing this is an old block that has been verified - no block goosip is needed
-  if (syncing_state_.is_syncing()) {
+void TaraxaCapability::onNewBlockVerified(DagBlock const &block, bool proposed) {
+  // If node is pbft syncing and block is not proposed by us, this is an old block that has been verified - no block
+  // goosip is needed
```

Comment:
if node syncing, node should not propose anything as it does not have current data ?

Reply:
I did this as a quick fix but I would propose completely remove the dag syncing state, if we are pbft synced, we are synced. There is always a chance that DAG structure is not exactly the same throughout network and it can be partitioned for some amount of time. Currently that constant call to restartSyncing from pbft manager puts us often into dag_syncing state which than would stop block gossip if I left it as it was before. This code is for gossiping and not proposing.

Reply:
if restartSyncing triggers only dag syncing, it does not block gossiping because syncing_state_.is_pbft_syncing() returns true only if we are syncing pbft blocks. And if we are truly pbft syncing, I think it should never happen that the node propose new dag/or pbft block so we should not get into this function with variables:  
- proposed == true and 
- syncing_state_.is_pbft_syncing() == true.

Reply:
So seems to me that changing condition from `if (syncing_state_.is_syncing())` to `if (syncing_state_.is_pbft_syncing())` might be enough and what you wanted to achieve by checking `!proposed` could be handled during proposing by checking pbft syncing and if we are syncing, nothing should be proposed ?

Reply:
there is a chance for race condition, we propose a block and only at that point we realize we are not pbft synced, we still should gossip it. Proposing is disabled by not feeding transactions to non-synced nodes but we could add additional protection